python/hsfs/core/feature_group_engine.py

```diff
@@ -259,10 +260,10 @@ def sql(self, query, feature_store_name, dataframe_type, online, read_options):
             read_options,
         )
 
-    def _update_features_metadata(self, feature_group, features):
+    def _update_features_metadata(self, feature_group: fg.FeatureGroup, features):
         # perform changes on copy in case the update fails, so we don't leave
         # the user object in corrupted state
-        copy_feature_group = fg.FeatureGroup.from_response_json(feature_group.to_dict())
+        copy_feature_group: fg.FeatureGroup = copy.deepcopy(feature_group)
```

Comment:
We should add `to_dict()` here instead https://github.com/logicalclocks/hopsworks-api/blob/main/python/hsfs/feature_group.py#L3557